src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSCopy.java

```diff
@@ -57,8 +62,36 @@ public void run(ActionContext context) throws IOException {
     }
     StorageClient storageClient = StorageClient.create(config.getProject(), config.getServiceAccount(),
                                                        isServiceAccountFilePath);
-    //noinspection ConstantConditions
-    storageClient.copy(config.getSourcePath(), config.getDestPath(), config.recursive, config.shouldOverwrite());
+
+    GCSPath destPath = config.getDestPath();
+    GCSPath undo = null;
+    Storage storage = storageClient.getStorage();
+
+    // create the destination bucket if not exist
+    Bucket bucket = null;
```

Comment:
this code is the same in GCSCopy and GCSMove. Can it be moved to the StorageClient so there isn't duplicate code?

Reply:
done